mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2496,18 +2496,22 @@ struct GridwiseGemmV2RewritePattern
 
       llvm::copy(tailResults, std::back_inserter(transformedTail));
     }
-    // Slice up vectors here to make it clearer that each store loop
-    // deals with a distinct set of values.
-    llvm::SmallVector<Value, 4> vectors;
-    vectors.reserve(transformedTail.size() * iterationsPerVectorC);
+
+    // Merge the vectors using miopen.insertslice and store loop to use
+    // miopen.extractslice so offset parsing can be deferred after loop
+    // unrolling.
+    int64_t vectorLen = vectorType.getShape()[0];
+    VectorType mergedType =
+        VectorType::get({static_cast<long>(vectorLen * transformedTail.size())},
```

Comment:
You might want the clone method on vectorType here?

Reply:
Can you give me more information about clone method? duplicating vector type and changing the shape still need same amount of code, I suppose.

Reply:
The `ShapedType`s have methods that, IIRC, let you do "I want this type but with a different shape (or datatype, or so on)"

Reply:
cloneWith just needs the same arguments, only shape is optional arg but this is changing shape.